Improve new referential integrity checker timeout tolerance

<!--
   For more information on how to populate this Task, see the PDS Wiki on User Story Development:
   https://github.com/NASA-PDS/nasa-pds.github.io/wiki/Issue-Tracking#user-story-development
-->

## 💡 Description
<!-- Enter description here. Make it detailed enough someone could actually know what you are doing, 
     but if you spend too much time on this, it probably deserves it's own story.
     
     NOTE: Be sure to add the sprint-backlog label if you plan on doing this right now.
-->
Intermittent Timeout when testing `validate-refs` throws an unhandled exception:

```
java.net.ConnectException: Timeout connecting to [search-geo-prod-6iz6lwiw6luyffpsq52ndsrtbu.us-west-2.es.amazonaws.com/<redacted>:443]
	at org.opensearch.client.RestClient.extractAndWrapCause(RestClient.java:953) ~[opensearch-rest-client-2.5.0.jar:2.5.0]
	at org.opensearch.client.RestClient.performRequest(RestClient.java:332) ~[opensearch-rest-client-2.5.0.jar:2.5.0]
	at org.opensearch.client.RestClient.performRequest(RestClient.java:320) ~[opensearch-rest-client-2.5.0.jar:2.5.0]
	at org.opensearch.client.RestHighLevelClient.internalPerformRequest(RestHighLevelClient.java:1911) ~[opensearch-rest-high-level-client-2.5.0.jar:2.5.0]
	at org.opensearch.client.RestHighLevelClient.performRequest(RestHighLevelClient.java:1877) ~[opensearch-rest-high-level-client-2.5.0.jar:2.5.0]
	at org.opensearch.client.RestHighLevelClient.performRequestAndParseEntity(RestHighLevelClient.java:1845) ~[opensearch-rest-high-level-client-2.5.0.jar:2.5.0]
	at org.opensearch.client.RestHighLevelClient.search(RestHighLevelClient.java:1088) ~[opensearch-rest-high-level-client-2.5.0.jar:2.5.0]
	at gov.nasa.pds.validate.ri.OpensearchDocument.search(OpensearchDocument.java:148) ~[validate-3.2.0-SNAPSHOT.jar:?]
	at gov.nasa.pds.validate.ri.OpensearchDocument.load(OpensearchDocument.java:67) [validate-3.2.0-SNAPSHOT.jar:?]
	at gov.nasa.pds.validate.ri.OpensearchDocument.exists(OpensearchDocument.java:157) [validate-3.2.0-SNAPSHOT.jar:?]
	at gov.nasa.pds.validate.ri.Cylinder.run(Cylinder.java:48) [validate-3.2.0-SNAPSHOT.jar:?]
	at gov.nasa.pds.validate.ri.Engine.processQueueUntilEmpty(Engine.java:72) [validate-3.2.0-SNAPSHOT.jar:?]
	at gov.nasa.pds.validate.ri.CommandLineInterface.process(CommandLineInterface.java:82) [validate-3.2.0-SNAPSHOT.jar:?]
	at gov.nasa.pds.validate.ReferenceIntegrityMain.main(ReferenceIntegrityMain.java:14) [validate-3.2.0-SNAPSHOT.jar:?]
Caused by: java.net.ConnectException: Timeout connecting to [search-geo-prod-6iz6lwiw6luyffpsq52ndsrtbu.us-west-2.es.amazonaws.com/44.238.156.194:443]
	at org.apache.http.nio.pool.RouteSpecificPool.timeout(RouteSpecificPool.java:169) ~[httpcore-nio-4.4.15.jar:4.4.15]
	at org.apache.http.nio.pool.AbstractNIOConnPool.requestTimeout(AbstractNIOConnPool.java:632) ~[httpcore-nio-4.4.15.jar:4.4.15]
	at org.apache.http.nio.pool.AbstractNIOConnPool$InternalSessionRequestCallback.timeout(AbstractNIOConnPool.java:898) ~[httpcore-nio-4.4.15.jar:4.4.15]
	at org.apache.http.impl.nio.reactor.SessionRequestImpl.timeout(SessionRequestImpl.java:198) ~[httpcore-nio-4.4.15.jar:4.4.15]
	at org.apache.http.impl.nio.reactor.DefaultConnectingIOReactor.processTimeouts(DefaultConnectingIOReactor.java:213) ~[httpcore-nio-4.4.15.jar:4.4.15]
	at org.apache.http.impl.nio.reactor.DefaultConnectingIOReactor.processEvents(DefaultConnectingIOReactor.java:158) ~[httpcore-nio-4.4.15.jar:4.4.15]
	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor.execute(AbstractMultiworkerIOReactor.java:351) ~[httpcore-nio-4.4.15.jar:4.4.15]
	at org.apache.http.impl.nio.conn.PoolingNHttpClientConnectionManager.execute(PoolingNHttpClientConnectionManager.java:221) ~[httpasyncclient-4.1.5.jar:4.1.5]
	at org.apache.http.impl.nio.client.CloseableHttpAsyncClientBase$1.run(CloseableHttpAsyncClientBase.java:64) ~[httpasyncclient-4.1.5.jar:4.1.5]
	at java.lang.Thread.run(Thread.java:832) ~[?:?]
```

On this failure, we should try 5 more times to reconnect and then throw a FATAL error.

--- a/src/main/java/gov/nasa/pds/validate/ri/CommandLineInterface.java
+++ b/src/main/java/gov/nasa/pds/validate/ri/CommandLineInterface.java
@@ -47,6 +47,8 @@ public class CommandLineInterface {
   public int process(String[] args)
       throws IOException, ParseException, ParserConfigurationException, SAXException {
     int cylinders = 1;
+    CountingAppender counter = new CountingAppender();
+    Logger.getRootLogger().addAppender(counter);
     CommandLine cl = new DefaultParser().parse(this.opts, args);
     if (cl.hasOption('h')) {
       this.help();
@@ -72,7 +74,7 @@ public class CommandLineInterface {
         throw new ParseException("The thread count must be an integer greater than 0.");
       }
     } else
-      log.info("lidvids will be sequentially processed.");
+      this.log.info("lidvids will be sequentially processed.");
 
     this.log.info("Starting the reference integrity checks.");
     try {
@@ -83,20 +85,22 @@ public class CommandLineInterface {
       broken = engine.getBroken();
       total = engine.getTotal();
     } catch (IOException e) {
-      log.fatal("Cannot process request because of IO problem.", e);
+      this.log.fatal("Cannot process request because of IO problem.", e);
       throw e;
     } catch (ParserConfigurationException e) {
-      log.fatal("Could not parse the harvest configuration file.", e);
+      this.log.fatal("Could not parse the harvest configuration file.", e);
       throw e;
     } catch (SAXException e) {
-      log.fatal("Mal-formed harvest configuration file.", e);
+      this.log.fatal("Mal-formed harvest configuration file.", e);
       throw e;
     }
     if (-1 < this.total) {
       this.log.info("Summary:");
       this.log.info("   " + this.total + " products processed");
-      this.log.info("   " + this.broken + " errors");
-      this.log.info("   " + (cl.hasOption("A") ? "0" : "1") + " warnings");
+      this.log.info("   " + this.broken + " missing references");
+      this.log.info("   " + counter.getNumberOfFatals() + " fatals");
+      this.log.info("   " + (counter.getNumberOfErrors() - broken) + " errors not including missing references");
+      this.log.info("   " + counter.getNumberOfWarnings() + " warnings");
     }
     return this.broken == 0 ? 0 : 1;
   }

--- a/src/main/java/gov/nasa/pds/validate/ri/CountingAppender.java
+++ b/src/main/java/gov/nasa/pds/validate/ri/CountingAppender.java
@@ -1,0 +1,75 @@
+package gov.nasa.pds.validate.ri;
+
+import org.apache.log4j.Appender;
+import org.apache.log4j.Layout;
+import org.apache.log4j.Level;
+import org.apache.log4j.spi.ErrorHandler;
+import org.apache.log4j.spi.Filter;
+import org.apache.log4j.spi.LoggingEvent;
+
+class CountingAppender implements Appender {
+  private int err = 0, fatal = 0, warn = 0;
+  private ErrorHandler handler = null;
+  private Filter filter = null;
+  private Layout layout = null;
+  private String name = "";
+  @Override
+  public void addFilter(Filter newFilter) {
+    this.filter = newFilter;
+  }
+  @Override
+  public Filter getFilter() {
+    return this.filter;
+  }
+  @Override
+  public void clearFilters() {
+    this.filter = null;
+  }
+  @Override
+  public void close() {
+  }
+  @Override
+  public void doAppend(LoggingEvent event) {
+    Level level = event.getLevel();
+    if (level == Level.ERROR) this.err++;
+    if (level == Level.FATAL) this.fatal++;
+    if (level == Level.WARN) this.warn++;
+  }
+  @Override
+  public String getName() {
+    return this.name;
+  }
+  @Override
+  public void setErrorHandler(ErrorHandler errorHandler) {
+    this.handler = errorHandler;
+  }
+  @Override
+  public ErrorHandler getErrorHandler() {
+    return this.handler;
+  }
+  @Override
+  public void setLayout(Layout layout) {
+    this.layout = layout;
+  }
+  @Override
+  public Layout getLayout() {
+    return this.layout;
+  }
+  @Override
+  public void setName(String name) {
+    this.name = name;
+  }
+  @Override
+  public boolean requiresLayout() {
+    return false;
+  }
+  int getNumberOfErrors() {
+    return this.err;
+  }
+  int getNumberOfFatals() {
+    return this.fatal;
+  }
+  int getNumberOfWarnings() {
+    return this.warn;
+  }
+}

--- a/src/main/java/gov/nasa/pds/validate/ri/CountingAppender.java
+++ b/src/main/java/gov/nasa/pds/validate/ri/CountingAppender.java
@@ -31,9 +31,9 @@ class CountingAppender implements Appender {
   @Override
   public void doAppend(LoggingEvent event) {
     Level level = event.getLevel();
-    if (level == Level.ERROR) this.err++;
-    if (level == Level.FATAL) this.fatal++;
-    if (level == Level.WARN) this.warn++;
+    if (Level.ERROR.equals(level)) this.err++;
+    if (Level.FATAL.equals(level)) this.fatal++;
+    if (Level.WARN.equals(level)) this.warn++;
   }
   @Override
   public String getName() {

--- a/src/main/java/gov/nasa/pds/validate/ri/Cylinder.java
+++ b/src/main/java/gov/nasa/pds/validate/ri/Cylinder.java
@@ -2,8 +2,8 @@ package gov.nasa.pds.validate.ri;
 
 import java.util.ArrayList;
 import java.util.List;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
 
 public class Cylinder implements Runnable {
   final private AuthInformation registry;

--- a/src/main/java/gov/nasa/pds/validate/ri/Engine.java
+++ b/src/main/java/gov/nasa/pds/validate/ri/Engine.java
@@ -4,8 +4,8 @@ import java.util.ArrayDeque;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Queue;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
 
 public class Engine implements CamShaft {
   private final int cylinders;

--- a/src/main/java/gov/nasa/pds/validate/ri/OpensearchDocument.java
+++ b/src/main/java/gov/nasa/pds/validate/ri/OpensearchDocument.java
@@ -1,6 +1,7 @@
 package gov.nasa.pds.validate.ri;
 
 import java.io.IOException;
+import java.net.ConnectException;
 import java.net.URL;
 import java.security.KeyManagementException;
 import java.security.KeyStoreException;
@@ -20,8 +21,8 @@ import org.apache.http.impl.client.BasicCredentialsProvider;
 import org.apache.http.impl.nio.client.HttpAsyncClientBuilder;
 import org.apache.http.ssl.SSLContextBuilder;
 import org.apache.http.ssl.SSLContexts;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
 import org.opensearch.action.search.SearchRequest;
 import org.opensearch.action.search.SearchResponse;
 import org.opensearch.client.RequestOptions;
@@ -143,9 +144,18 @@ public class OpensearchDocument implements DocumentInfo, RestClientBuilder.HttpC
      * Therefore cannot simply extend it or implement a common interface to do a more subtle
      * insertion of unit testing.
      */
-    if (OpensearchDocument.sourceOverride != null)
-      return OpensearchDocument.sourceOverride.search(client, request);
-    return client.search(request, RequestOptions.DEFAULT);
+    int iteration = 0;
+    while (true) {
+      try {
+        if (OpensearchDocument.sourceOverride != null)
+          return OpensearchDocument.sourceOverride.search(client, request);
+        return client.search(request, RequestOptions.DEFAULT);
+      } catch (ConnectException ce) {
+        iteration++;
+        if (iteration < 5) this.log.warn("Could not connect but trying again: " + iteration);
+        else throw ce;
+      }
+    }
   }
 
   public OpensearchDocument(AuthInformation context) {

